packages/@aws-cdk/service-spec-types/src/types/resource.ts

```diff
@@ -494,6 +494,19 @@ export class RichPropertyType {
     }
   }
 
+  /**
+   * Whether the current type is assignable to the RHS type.
+   *
+   * This is the same as normal equality, except when the RHS type is a union and the LHS is not a union.
+   */
+  public assignableTo(rhs: PropertyType): boolean {
+    if (this.type.type != 'union' && rhs.type === 'union') {
+      return rhs.types.some((t) => this.equals(t));
```

Comment:
I think this is not right? Don’t we need to check whether lhs is a subset of rhs?